handsontable/src/pluginHooks.js

```diff
@@ -1405,6 +1425,26 @@ const REGISTERED_HOOKS = [
    */
   'afterColumnMove',
 
+  /**
+   * Fired by {@link ManualColumnFreeze} plugin before freezing a column.
+   *
+   * @event Hooks#beforeColumnUnfreeze
+   * @since 12.1.0
+   * @param {number} column Visual column index of unfrozen column.
+   * @param {boolean} unfreezePerformed Indicates if order of columns will be changed by unfreeze.
```

Comment:
```suggestion
   * @param {boolean} unfreezePerformed Indicates if order of columns will be changed by unfreeze.
   * @returns {boolean|undefined} If `false` the column will not be unfrozen, `true` otherwise.
```